File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/elementlist/constructor/EmptyTagListConstructor.java
package com.tickaroo.tikxml.annotationprocessing.elementlist.constructor;

import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Xml;
import com.tickaroo.tikxml.annotationprocessing.element.EmptyTag;
import java.util.List;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class EmptyTagListConstructor {


  private List<EmptyTag> tags;

  public EmptyTagListConstructor(
      @Element List<EmptyTag> tags) {
    this.tags = tags;
  }

  public List<EmptyTag> getTags() {
    return tags;
  }
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/elementlist/constructor/EmptyTagListConstructor.kt
package com.tickaroo.tikxml.annotationprocessing.elementlist.constructor

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml
import com.tickaroo.tikxml.annotationprocessing.element.EmptyTag

/**
 * @author Hannes Dorfmann
 */
@Xml
class EmptyTagListConstructor(
        @param:Element val tags: List<EmptyTag>
)
